### ISSUE TITLE
Incomplete Beta Inverse

### ISSUE THREAD

**Issue by spinkney** [stan-dev/math#2513]
## Description
Add incomplete beta inverse. This is needed for the quantile function of the beta distribution (if or when quantile functions get added). By having this available to users they can define the [quantile function of the beta distribution](https://github.com/mborland/math/blob/a058883c71174be7fe87c0a2d79e86c0310e0ffb/include/boost/math/distributions/beta.hpp#L494) as well as the quantile function of the [skewed generalized t distribution](https://github.com/cran/sgt/blob/8bb840480ea45b48aee6dca1b06a9d6453ad0243/R/InternalSGT.R#L30)

The function is available in boost at https://www.boost.org/doc/libs/1_74_0/libs/math/doc/html/math_toolkit/sf_beta/ibeta_inv_function.html

The derivative is defined at https://functions.wolfram.com/GammaBetaErf/InverseBetaRegularized/20/01/01/

**Comment by andrjohns** [comment]
Im actually planning on working on through the quantile functions (starting with the Gamma distribution at the moment), so this is already on my to-do list (unless you've already started!)

**Comment by spinkney** [comment]
Nice! I have some icdfs at https://github.com/spinkney/helpful_stan_functions/tree/main/functions%2Finverse_cdf. 

If the quantile functions are getting implemented, I think I'll start a discussion on copula functions and their design.

**Comment by spinkney** [comment]
Do you know if there's any discussion on a reserved name for these? Should it be icdf or quantile?

**Comment by andrjohns** [comment]
I was planning on proposing using the `_icdf` suffix, so that it's all consistent with the lpdf/lcdf/lccdf naming scheme, but no actual discussion yet

**Comment by spinkney** [comment]
@andrjohns is it possible to _also_ expose this function when you add the icdfs? This functionay be useful for user defined icdfs in the future

**Comment by andrjohns** [comment]
Yep! The plan is to expose all of the intermediary functions. This function is just waiting on the Hypergeometric pFq, which is needed for the derivatives wrt a and b

**Comment by spinkney** [comment]
Well, I found an implementation that wasn't too hard to write in Stan. The stan code is at https://github.com/spinkney/helpful_stan_functions/blob/main/functions/special/inc_beta_inverse.stan.

I checked the output on  a few cases compared to qbeta in R.  Not sure if you want to test against boosts version or not.

**Comment by spinkney** [comment]
@andrjohns there's a wealth of inverse cdf's here https://people.sc.fsu.edu/~jburkardt/cpp_src/cpp_src.html. I'm going to put some together in Stan but can start trying to get some into the cpp. I want to get this beta one in soon as it allows the beta, student_t, skew_student_t, and probably others right off the bat.

@bgoodri has made a [good case](https://github.com/stan-dev/design-docs/pull/42) for `_qf` as the suffix, especially if we start implementing the derivative of those (which we should!) and then call them `_dqf` and `_ldqf`. In addition, I've already added syntax highlighting for the `_qf` in the highlights.js. 

Look at this student_t_qf:
```r
# http://www.homepages.ucl.ac.uk/~ucahwts/lgsnotes/JCF_Student.pdf
qt2 <- function(u, nu){
  x <- ifelse (u < 0.5, 2 * u, 2 * (1 - u))
  sign(u - 0.5) * sqrt( nu * (1 / (qbeta(x, shape1=nu / 2, shape2= 0.5))) - 1)
}

all.equal(qt2(0.2, 1), qt(0.2, 1))
```

**Comment by andrjohns** [comment]
Oh excellent, thanks for prompting me on this! Will get into having a look at those
